server/calendar/src/scrape_task/tumonline_calendar_connector.rs

```diff
@@ -16,61 +14,39 @@ enum RequestStatus {
     Success(String),
     Timeout,
     NotFound,
-    TooLarge,
     OneDayFaulty,
     Error,
 }
-async fn request_body(client: &Client, url: String) -> RequestStatus {
-    let req = client.get(&url).send().await;
+
+async fn request_body(url: String) -> RequestStatus {
+    let req = reqwest::get(&url).await;
     let body = match req {
-        Ok(mut res) => match res.status().as_u16() {
-            200 => res.body().limit(2_usize.pow(32)).await,
+        Ok(res) => match res.status().as_u16() {
+            200 => res.text().await,
```

Comment:
Is the limit (`res.body().limit(2_usize.pow(32))` beforehand) no longer required for reqwest or just no longer in general (bc 2³² is a lot anyway)?

Reply:
`reqwest` has a different API for specifying if a limit is required.
We want to scrape indiscriminately (requests are smaller than memory => not an issue)

see https://www.reddit.com/r/rust/comments/mkk1eb/comment/gtg9jox/

[Streaming the text](https://docs.rs/reqwest/latest/reqwest/struct.Response.html#method.text) could be a oppertunity to save this memory, but would require a different processing approach.
The current approach works => I dont think we need to improve this